src/Hazelcast.Net/Clustering/ClusterMembers.cs

```diff
@@ -328,8 +333,9 @@ public async Task RemoveConnectionAsync(MemberConnection connection)
             }
             finally
             {
-                // don't forget to resume the queue
-                _memberConnectionQueue?.Resume(drain);
```

Comment:
Verifying my understanding. We disabled the whole member connection queue since client is not connected but we know that when client disconnected, reconnect will be triggered  (by cient state changed event), right?

Reply:
When the client is disconnected.... we will enter the "first connection" process again, ie we'll try to establish a new connection to the cluster again. if that succeeds... the new connection will be added to members, we'll realize we have a member <--> connection match, we'll deduce that we are connected - and *then* we'll resume the connection queue so it handles all the other members and constantly make sure that all members are connected.

when we lose 1 connection... if it's one amongst others (there still are some members connected) the queue will take care of it: the member is queued, and eventually re-connected.

OTOH if we lose the last connection... we go disconnected and we stop the queue, because we're going to start connecting the "first member" and we cannot have a parallel process *also* connecting members - because the "first member" is special.